packages/dom/src/node.rs

```diff
@@ -797,6 +800,50 @@ impl Node {
                 y,
             }))
     }
+
+    /// Scroll this node by given x and y
+    /// Will bubble scrolling up to parent node once it can no longer scroll further
+    pub fn scroll_by(&mut self, x: f64, y: f64) {
+        self.scroll_offset.x += x;
+        self.scroll_offset.y += y;
+
+        //If we're past our scroll bounds, transfer scrolling to parent
+        if self.scroll_offset.x < 0.0
+            || self.scroll_offset.x > self.final_layout.scroll_width() as f64
+            || self.scroll_offset.y < 0.0
+            || self.scroll_offset.y > self.final_layout.scroll_height() as f64
+        {
+            if let Some(parent) = self.parent {
+                unsafe {
```

Comment:
If you moved the scroll_by method to the `Document` then you could avoid the `unsafe` here.